compiler/src/intermediate_representation/store_bucket.rs

```diff
@@ -172,20 +177,22 @@ impl StoreBucket{
         if let AddressType::SubcmpSignal { cmp_address, .. } = &dest_address_type {
             let addr = cmp_address.produce_llvm_ir(producer).expect("The address of a subcomponent must yield a value!");
             let counter = producer.template_ctx().load_subcmp_counter(producer, addr);
-            let value = create_load_with_name(producer, counter, "load.subcmp.counter");
-            let new_value = create_sub_with_name(producer, value.into_int_value(), create_literal_u32(producer, 1), "decrement.counter");
-            assert_eq!(1, context.size, "unhandled array store");
-            create_store(producer, counter, new_value);
+            if !counter.is_null() {
```

Comment:
For example here, it's not clear why are you checking that is a null pointer because you have situations where you don't have that data. I think it will be more clean to use an optional instead and not tie the absence of data to the llvm API